src/core/lib/security/context/security_context.h

```diff
@@ -41,6 +41,16 @@
 
 extern grpc_core::DebugOnlyTraceFlag grpc_trace_auth_context_refcount;
 
+/* --- grpc_security_context_extension ---
+
+   Extension to the security context that may be set in a filter and accessed
+   later by a higher level method on a grpc_call object. */
+
+struct grpc_security_context_extension {
```

Comment:
This C type is actually for a different purpose, and I don't think we should use it here.  You can move this back down to its original location, below `grpc_auth_context`.

For this PR, let's instead introduce an idiomatic C++ type, like this:

```
struct grpc_auth_context
    : public grpc_core::RefCounted<grpc_auth_context,
                                   grpc_core::NonPolymorphicRefCount> {
 public:
  // A base class for extensions to inherit from.
  class Extension {
   public:
    virtual ~Extension() = default;
  };

  void set_extension(std::unique_ptr<Extension> extension) {
    extension_ = std::move(extension);
  }

  // ...

 private:
  // ...
  std::unique_ptr<Extension> extension_;
};
```

Then you can define a subclass of `grpc_auth_context::Extension` that contains whatever you need it to.